docs/application/native/guides/machine-learning/machine-learning-training.md

```diff
@@ -0,0 +1,402 @@
+# Machine Learning Training
+
+Machine Learning Training allows to construct, control and train a machine learning model in Tizen devices.
+
+Please note that:
+
+* This feature remains experimental for Tizen 6.0 and is not full-fledged yet for every needs.
+* Every example code is not fully handling error for brevity. Error should be handled more extensively then example code written here.
+
+## Prerequisites
+
+To enable your application to use the machine learning training functionality:
+
+1. To use the functions and data types of the Machine Learning Inference API, include the `<nntrainer.h>` header file in your application:
+
+   ```c
+   #include <nntrainer.h>
+   ```
+
+2. To use the Machine Learning Training API, include the following features in your `tizen-manifest.xml` file:
+
+   ```xml
+   <feature name="http://tizen.org/feature/machine_learning">true</feature>
+   <feature name="http://tizen.org/feature/machine_learning.training">true</feature>
+   ```
+
+3. If you want to save/load model files from outside, the application has to request permission by adding following privileges to the `tizen-manifest.xml` file:
+
+  ```xml
+  <privileges>
+    <!-- For accessing media storage -->
+    <privilege>http://tizen.org/privilege/mediastorage</privilege>
+    <!-- For accessing external storage -->
+    <privilege>http://tizen.org/privilege/externalstorage</privilege>
+  </privileges>
+  ```
+
+## Building Blocks
+
+There are 4 major components you need to get familiar with. With those components, you can prepare a model and run.
+
+### Model
+
+![model](./media/model.png)
+
+Model is a wrapper component that has topology of layers, optimizers, datasets.
+Model conducts training and update parameters that can later be used for inference.
```

Comment:
Model performs training and saves the updated parameters ....

Reply:
Done